composer/utils/object_store/uc_object_store.py

```diff
@@ -203,12 +206,27 @@ def get_object_size(self, object_name: str) -> int:
     def list_objects(self, prefix: Optional[str]) -> List[str]:
         """List all objects in the object store with the given prefix.
 
+         .. note::
+
+            This function removes the directories from the returned list.
+
         Args:
             prefix (str): The prefix to search for.
 
         Returns:
             list[str]: A list of object names that match the prefix.
         """
-        # TODO: Implement this function once UC volumes list endpoint is available in the SDK
-        del prefix  # unused
-        raise NotImplementedError(f'{type(self).__name__}.list_objects is not implemented')
+        if not prefix:
+            prefix = self.prefix
+
+        from databricks.sdk.core import DatabricksError
+        try:
+            data = json.dumps({'path': self._get_object_path(prefix)})
+            # NOTE: This API is in preview and should not be directly used outside of this instance
+            resp = self.client.api_client.do(method='GET',
+                                             path=self._UC_VOLUME_LIST_API_ENDPOINT,
+                                             data=data,
+                                             headers={'Source': 'mosaicml/composer'})
```

Comment:
@seansmith-db here is how we are using the list API. let me know what headers we need to pass for the correct attribution. We cannot pass `User-Agent` because that's overwritten by the API here: https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/core.py#L1141

Reply:
Spoke with Sean offline on slack. Given that `User-Agent` does not work, this alternative header should work fine to identify traffic from composer.